AgentDiscoveries-Backend/src/main/java/org/softwire/training/api/core/PermissionsVerifier.java

```diff
@@ -0,0 +1,27 @@
+package org.softwire.training.api.core;
+
+import org.softwire.training.api.models.ErrorCode;
+import org.softwire.training.api.models.FailedRequestException;
+import org.softwire.training.db.daos.AgentsDao;
+import org.softwire.training.db.daos.UsersDao;
+import org.softwire.training.models.User;
+
+import javax.inject.Inject;
+import java.util.Optional;
+
+public class PermissionsVerifier {
+    private final UsersDao usersDao;
+
+    @Inject
+    public PermissionsVerifier(UsersDao usersDao) {
+        this.usersDao=usersDao;
+    }
+
+    public boolean isAdmin(int userId) throws FailedRequestException {
+        Optional<User> optionalUser = usersDao.getUser(userId);
+        if (!optionalUser.isPresent()) {
+            throw new FailedRequestException(ErrorCode.NOT_FOUND, "user wasn't found");
+        }
+        return optionalUser.get().isAdmin();
```

Comment:
Optionals can be used in a slightly more functional way using map/ifPresent/orElse etc. In this particular case:

```
return optionalUser.map(User::isAdmin)
    .orElseThrow(() => new FailedRequestException(ErrorCode.NOT_FOUND, "User not found");
```

Reply:
Ah, nice :)